test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,275 @@
+package seedu.addressbook.data;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import seedu.addressbook.data.exception.ConstraintViolationException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.data.tag.UniqueTagList.DuplicateTagException;
+import seedu.addressbook.data.tag.UniqueTagList.TagNotFoundException;
+
+import static seedu.addressbook.util.TestUtil.isEmpty;
+import static seedu.addressbook.util.TestUtil.isIdentical;
+import static seedu.addressbook.util.TestUtil.getAllTags;
+
+public class AddressBookTest {
+    private Tag tagPrizeWinner;
+    private Tag tagScientist;
+    private Tag tagMathematician;
+    private Tag tagEconomist;
+
+    private Person aliceBetsy; // has tagMathematician
+    private Person bobChaplin; // has tagMathematician
+    private Person charlieDouglas; // has tagScientist
+    private Person davidElliot; // has tagEconomist, tagPrizeWinner
```

Comment:
The info in these comments are also present in another place (in the `setUp` method), and therefore this is a violation of the DRY principle. Comments can be removed.